lib/generators/logidze/install/functions/logidze_capture_exception.sql

```diff
@@ -0,0 +1,23 @@
+CREATE OR REPLACE FUNCTION logidze_capture_exception(error_data jsonb) RETURNS boolean AS $body$
+  -- version: 1
+BEGIN
+  -- Feel free to change this function to change Logidze behavior on exception.
+  --
+  -- Return `false` to raise exception or `true` to commit record changes.
+  --
+  -- `error_data` contains:
+  --   - returned_sqlstate
+  --   - message_text
+  --   - pg_exception_detail
+  --   - pg_exception_hint
+  --   - pg_exception_context
+  --   - schema_name
+  --   - table_name
+  -- Learn more about available keys:
+  -- https://www.postgresql.org/docs/9.6/plpgsql-control-structures.html#PLPGSQL-EXCEPTION-DIAGNOSTICS-VALUES
```

Comment:
to your mind, how might error handling different when `(coalesce(current_setting('logidze.full_snapshot', true), '') = 'on')`? Asking since it's used in the caller and available in this context, though not explicitly passed in.

Reply:
Honestly didn't understand your thought. How is the setting should affect this "global" error handler?

Reply:
Here's my thought process.

1. logidze_logger is called
2. `IF (coalesce(current_setting('logidze.full_snapshot', true), '') = 'on') THEN` save log
3. `ELSE` save log
4. EXCEPTION (non-NUMERIC_VALUE_OUT_OF_RANGE) then `logidze_capture_exception(err_jsonb)`


I noticed that there are two possible paths to save log data and the error handler doesn't pass in the distinguishing condition between the two paths (logidze.full_snapshot). 

It's also not obvious to me how the handler might need to behave differently depending on which branch raised the exception, but I thought it worth asking since you have a better understanding of the library (and probably pg in general) than I do.

From your answer I infer it's unlikely to make a difference in error handling.

Reply:
Thanks for the clarification =) I think one can check setting `(coalesce(current_setting('logidze.full_snapshot', true), '') = 'on')` in the error handler, so it should do the trick.